### PR TITLE
WAITP-1452: Datatrak-web refactor survey context

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Refactor Survey cancel modal to be generic
- Move SurveyContext down to be in the survey page

### CODE REVIEW

File: packages/datatrak-web/src/api/mutations/useLogin.ts [hunk 1]
```diff
@@ -12,12 +12,19 @@ type LoginCredentials = {
...
+    return typeof state.from === 'string';
+  }
+  return false;
+}
 export const useLogin = () => {
   const queryClient = useQueryClient();
   const navigate = useNavigate();
   const location = useLocation();
 
-  const { from } = location.state as { from: string };
```
Comment (tcaiger) [inline packages/datatrak-web/src/api/mutations/useLogin.ts:20]: I was getting an error on first login where state was null so I added some more robust checking.

Comment (acdunham) [reply packages/datatrak-web/src/api/mutations/useLogin.ts:20]: We could probably simplify this to be just `location.state && location.state.from`

File: packages/datatrak-web/src/layout/Header.tsx [hunk 2]
```diff
@@ -47,27 +46,30 @@ const LogoButton = styled(Button)`
...
 export const Header = () => {
-  const { surveyCode } = useParams();
-  const { openCancelConfirmation, isSuccessScreen } = useSurveyForm();
-  const isActiveSurvey = !!surveyCode && !isSuccessScreen;
-  const onClickLogo = () => {
-    if (surveyCode && !isSuccessScreen) {
-      openCancelConfirmation();
+  const [surveyCancelModalIsOpen, setIsOpen] = useState(false);
+  const isSurveyScreen = !!useMatch(ROUTES.SURVEY_SCREEN);
+  const isSuccessScreen = !!useMatch(ROUTES.SURVEY_SUCCESS);
```
Comment (acdunham) [inline packages/datatrak-web/src/layout/Header.tsx:51]: Nice, I like this

**Review by acdunham — APPROVED** [review]
Looks good, thanks for doing that. A couple small comments but nothing major

File: packages/datatrak-web/src/api/mutations/useLogin.ts (hunk 1 above)
Comment (tcaiger) [reply packages/datatrak-web/src/api/mutations/useLogin.ts:20]: I tried that but I get a TS error. I could cast it using `as` but Rohan said that it's not good practise to cast and that I should use typeguards so I'm trying that out. It seems a bit verbose so maybe there is a more concise way of doing it.

**Pull request merged**